### DISCUSSION TITLE
maybe a pull request?

### DISCUSSION THREAD

**Discussion by spyros-tsimboulis** [nlohmann/json#2910] [Ideas]
hi,
I have created a repo at [https://github.com/Spyros-DC/from_to_json](url) , for json serialization - deserialization on top of nlohmann json. Some points that I find interesting are:

- it's simple to use
- can handle objects of arbitrary depth
- it works at compile time

I have some examples at my repo at main function. I think that has many advantages over nlohmann json serialization - deserialization. Can you take a look?

Thanks for your time, best regards,
Spyros

**Comment by nlohmann** [comment]
I do not fully understand what your project does. From a brief look, I see that you are using C++17 and Boost - both are not compatible with this library which targets C++11 and uses only the STL. Can you elaborate?

**Comment by spyros-tsimboulis** [comment]
The target is actually C++14, with some refactoring I believe I can target C++11. Boost it's not a part of this, I added it just to show how to deal with types of other libraries. 

In a nutshell, the repo is a combination of :

- ideas borrowed from CppCon 2014 Walter E. Brown "Modern Template Metaprogramming: A Compendium"

- the C++ concept of " pointer to member "

- the simple structure of json objects

In order to use it, in the C++ class / struct we add a public static constexpr member " properties ". Then we can call from - to json functions that works in a natural way and in an arbitrary depth and combination of objects.

**Reply by nlohmann** [reply]
So the difference is that client code defines these properties rather than a `to_json` function or a `NLOHMANN_DEFINE_TYPE_INTRUSIVE` macro?

**Reply by spyros-tsimboulis** [reply]
apologies for this... i thought ( by my mistake ) that nlohmann lib was going only one level deep. so nothing really essential to add by my repo ...BTW please accept my appreciation for your library, we use it a lot, and again thank you for your time!
